### PR TITLE
Feature/rename meta to viewmodel

### PR DESCRIPTION
**Pull request by martingagnon**
## Description
Make major change in the library and rename MetaViews to ViewModels.

#### Common
- Rename `MetaView` -> `ViewModel`, `MetaLabel` -> `LabelViewModel`, `MetaButton` -> `ButonViewModel` (and so on...)
- Rename `MetaSelector` -> `StateSelector`
- Rename `MetaImageResourceProvider` -> `ImageViewModelResourceProvider`
- Rename `MetaAction` -> `ViewModelAction`
- Rename `MetaView.onTap` -> `ViewModel.action`
- Rename `MetaInputType` -> `InputTextType`

(Probably forgot some other stuff to tell, but no meta should be left over)

#### android-ktx
- Rename `meta_view` binding property to `view_model`

#### iOS
- Rename `UIView.metaView` -> `UIView.viewModel`, `UILabel.metaLabel` -> `UILabel.labelViewModel` (and so on...)

After this PR, the repository will also be renamed.

## Motivation and Context
At Mirego, we wanted to use a naming that will communicate clearly to the community what it is doing.

## How Has This Been Tested?
- [X] All features has been added/updated in sample application ( /sample )

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: android-ktx/src/main/java/com/mirego/trikot/viewModels/ButtonViewModelBinder.kt [hunk 1]
```diff
@@ -1,32 +1,32 @@
-package com.mirego.trikot.metaviews
+package com.mirego.trikot.viewModels
```
Comment (gbourassa) [inline android-ktx/src/main/java/com/mirego/trikot/viewModels/ButtonViewModelBinder.kt]: En général je dirais que les nom de package sont au singulier.... 😅

File: android-ktx/build.gradle [hunk 2]
```diff
@@ -40,7 +40,7 @@ configurations.all {
 
 dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
-    api 'com.mirego.trikot:metaviews:0.47.1'
+    api 'com.mirego.trikot:viewModels:0.48.1-SNAPSHOT'
```
Comment (hugolefrancois) [inline android-ktx/build.gradle]: Attention au -SNAPSHOT

File: jenkins-jobs/plugin_release.groovy [hunk 3]
```diff
@@ -32,7 +32,7 @@ job(context.jobFullName) {
         gradle {
             useWrapper()
             makeExecutable()
-            tasks(':metaviews:release')
+            tasks(':viewModels:release')
```
Comment (hugolefrancois) [inline jenkins-jobs/plugin_release.groovy]: view**m**odels

File: sample/android/build.gradle [hunk 4]
```diff
@@ -96,7 +96,7 @@ dependencies {
     implementation 'androidx.lifecycle:lifecycle-reactivestreams-ktx:2.0.0'
 
     implementation "com.mirego.trikot:android-ktx:$trikot_streams_android_ktx_version"
-    implementation "com.mirego.trikot.metaviews:android-ktx:$trikot_metaviews_android_ktx_version"
+    implementation "com.mirego.trikot.viewModels:android-ktx:$trikot_viewmodels_android_ktx_version"
```
Comment (hugolefrancois) [inline sample/android/build.gradle]: view**M**odels

File: sample/android/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:tools="http://schemas.android.com/tools"
-    package="com.trikot.metaviews.sample">
+    package="com.trikot.viewModels.sample">
```
Comment (hugolefrancois) [inline sample/android/src/main/AndroidManifest.xml]: view**m**odels

File: sample/android/src/main/java/com/trikot/viewModels/sample/ListActivity.kt [hunk 6]
```diff
@@ -1,15 +1,15 @@
-package com.trikot.metaviews.sample
+package com.trikot.viewModels.sample
```
Comment (hugolefrancois) [inline sample/android/src/main/java/com/trikot/viewModels/sample/ListActivity.kt]: pas de majuscule dans les nom de package. J'arrête de les flagger 😜

File: sample/android/src/main/java/com/trikot/viewModels/sample/databinding/MetaListItemAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package com.trikot.viewModels.sample.databinding
+
+import androidx.databinding.BindingAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.trikot.viewModels.sample.R
+import com.trikot.viewModels.sample.viewModels.*
+
+class MetaListItemAdapter :
```
Comment (hugolefrancois) [inline sample/android/src/main/java/com/trikot/viewModels/sample/databinding/MetaListItemAdapter.kt]: Encore le mot Meta dans le nom ici 🤔

File: android-ktx/src/main/java/com/mirego/trikot/viewModels/ButtonViewModelBinder.kt (hunk 1 above)
Comment (martingagnon) [reply android-ktx/src/main/java/com/mirego/trikot/viewModels/ButtonViewModelBinder.kt]: As its the library name, it seems to me that it should be plural, like trikot.streams

File: sample/gradle.properties [hunk 8]
```diff
@@ -3,16 +3,16 @@ kotlin_version = 1.3.70
...
 trikot_streams_version=0.56.1
-trikot_metaviews_version=0.47.1
+trikot_viewmodels_version=0.48.1-SNAPSHOT
 trikot_http_version=0.19.1
 trikot_datasources_version=0.15.1
 trikot_graphql_version=0.17.1
 trikot_kword_version=0.8.1
 trikot_streams_android_ktx_version=0.31.1
-trikot_metaviews_android_ktx_version=0.53.1
+trikot_viewmodels_android_ktx_version=0.54.1-SNAPSHOT
```
Comment (hugolefrancois) [inline sample/gradle.properties]: -SNAPSHOT 👀

File: swift-extensions/UIButtonExtensions.swift [hunk 9]
```diff
@@ -199,15 +176,15 @@ extension UIButton {
...
             self?.updateBackgroundColor(selector)
         }
     }
 
     @objc
     private func onButtonTouchUp() {
-        guard let tapAction = tapAction else { return }
-        tapAction.execute(actionContext: self ?? nil)
+        guard let buttonViewModel = buttonViewModel else { return }
+        observe(buttonViewModel.action.first()) {[weak self] (value: ViewModelAction) in value.execute(actionContext: self ?? nil) }
```
Comment (hugolefrancois) [inline swift-extensions/UIButtonExtensions.swift]: T'as enlevé la modif de Marc? Le first est un problème à cause de l'operator startWith qui contient un buffer de value dans le publisher non.

Comment (martingagnon) [reply swift-extensions/UIButtonExtensions.swift]: I'll fix it!

File: android-ktx/build.gradle (hunk 2 above)
Comment (martingagnon) [reply android-ktx/build.gradle]: on it!

File: jenkins-jobs/plugin_release.groovy (hunk 3 above)
Comment (martingagnon) [reply jenkins-jobs/plugin_release.groovy]: on it!

**Pull request merged**